### ISSUE TITLE
Duplicate props in Select API docs

### ISSUE THREAD

**Issue by urlDev** [JedWatson/react-select#4889]
After checking the issues and PRs, I decided its best to open an issue ticket about this.

Currently, `props` in Select API docs have many duplicates, can be seen under [select sub menu](https://react-select.com/props#select-props).

### Select props
![image](https://user-images.githubusercontent.com/56873744/139577491-654d7f8b-da0a-4951-b61e-fdd5347fc775.png)

### Duplicates (start after `tabSelectsValue`)
![image](https://user-images.githubusercontent.com/56873744/139577430-d2c59979-c2a8-4dc5-b109-61712a1d4e9b.png)

**Comment by shavidze** [comment]
Is it still an issue? How is it possible to assign it to me? I see inside the project and I believe it comes from https://github.com/Thinkmill/magical-types/tree/master/packages/pretty/src/pretty-proptypes this project, so the issue belongs to it

**Comment by ebonow** [comment]
> Is it still an issue? How is it possible to assign it to me?

Yes it is still an issue and we would happily accept a PR for this.

Not sure if you are the person to ask about this, but I had a few other questions with the magical-types/pretty-prototypes and was also wondering what additional customization might be possible with the library specifically in regards to printing default values, editing the styling, or omitting specified prop types. 

But most of all thank you in advance for any help you can provide for this.

**Comment by Rall3n** [comment]
@ebonow  @shavidze I don't think the problem lies with `@magical-types`, but instead in our composition of `StateManagerProps` (exported as `Props`).

https://github.com/JedWatson/react-select/blob/2d3d6afe18421ea5b6c3f8f0db479ad323c35639/packages/react-select/src/useStateManager.ts#L11-L38

Looking at the types we can see that `SelectPropsWithOptionalStateManagedProps` (and therefore `StateManagerProps`) basically gets 2 sets of `PublicBaseSelectProps`: One without `StateManagedPropKeys` and one with all as optional (E.g. `inputValue` does not occur twice in the documentation.). `@magical-types` does not seem to remove duplicate props by default.

So we either change which props to use for this particular section of documentation, see if `@magical-types` does allow removal of duplicates, or we rethink the composition of `SelectPropsWithOptionalStateManagedProps`.
